src/chainrules.jl

```diff
@@ -150,3 +150,40 @@ function ChainRulesCore.rrule(::typeof(ifftshift), x::AbstractArray, dims)
     end
     return y, ifftshift_pullback
 end
+
+# plans
+function ChainRulesCore.frule((_, _, Δx), ::typeof(*), P::Plan, x::AbstractArray) 
+    y = P * x 
+    Δy = P * Δx
+    return y, Δy
+end
+function ChainRulesCore.rrule(::typeof(*), P::Plan, x::AbstractArray)
+    y = P * x
+    project_x = ChainRulesCore.ProjectTo(x)
+    Pt = P'
+    function mul_plan_pullback(ȳ)
+        x̄ = project_x(Pt * ȳ)
+        return ChainRulesCore.NoTangent(), ChainRulesCore.NoTangent(), x̄
+    end
+    return y, mul_plan_pullback
+end
+
+function ChainRulesCore.frule((_, ΔP, Δx), ::typeof(*), P::ScaledPlan, x::AbstractArray) 
+    y = P * x 
+    Δy = P * Δx + ΔP.scale / P.scale * y
+    return y, Δy
+end
+function ChainRulesCore.rrule(::typeof(*), P::ScaledPlan, x::AbstractArray)
+    y = P * x
+    project_x = ChainRulesCore.ProjectTo(x)
+    project_scale = ChainRulesCore.ProjectTo(P.scale)
```

Comment:
This is just a number, isn't it? IIRC typically `ProjectTo` is not used for numbers.
```suggestion
```

Reply:
Since `scale` can either be `Real` or `Complex`, this seems necessary for correctness so that we can project in the former case. This looks like a similar example of projecting a number: https://github.com/JuliaDiff/ChainRules.jl/blob/fef6aa821e871db5df11061312ba02929b7fa16d/src/rulesets/Base/fastmath_able.jl#L198. Please let me know if there may be a better option for this case, though

Reply:
Ah good point, I didn't think about projection to real numbers.